### ISSUE TITLE
Add a callback function parameter to requestOpenExtension function

### ISSUE THREAD

**Issue by zhangusc** [Adobe-CEP/CEP-Resources#56]
The current requestOpenExtension() function is non-blocking call but doesn't provide a callback parameter. There is no reliable way to run certain actions right after the extension get started. A callback parameter needs to be added to the function.

**Comment by arthurwei1975** [comment]
You can use another way to achieve this.
- In extension 1, you call requestOpenExtension() to open extension 2.
- Extension 2 sends a CEP event to extension 1.
- When extension 1 receives the event, you know extension 2 has been launched, and the event callback is the call back you need.

**Comment by zhangusc** [comment]
That's exact the workaround we are currently using. But the workflow is buggy since each extension has its own life-cycle and it is controlled by the host application. It is very tricky for extension 2 to fire the event. When should extension 2 fire the event. Extension 2 can be started up by the host application (no need to fire an event) or started up by extension 1 via requestOpenExtension() function (need fire the event). We eventually got it working after fixing a bunch of problems.

A callback parameter on requestOpenExtension() function can solve all those problems and make the workflow clean and bug-free.

**Comment by zhangusc** [comment]
@sujaisivanandan could you update the community the reason of closing the issue, fixed, not in the plan, as design, etc.?

**Comment by pixels4nickels** [comment]
Yes, please don’t just close issues without making the reason transparent

**Comment by mxdmly** [comment]
> 
I think you need to check if there is an error in the line "var evt = new CSXSEvent();"

**Comment by Inventsable** [comment]
Just as a note, you can actually programmatically open any extension from JSX if you're working in ILST or PHXS if you have access to Actions with Insert Menu Command (or equivalent run menu command methods). For instance, if I record an Action in ILST then Insert Menu Command to the name of the CEP extension, it will open that CEP extension. I can export it and save that aia text as a string then dynamically load it, run the Action, unload and return to my evalScript callback.

So in the case above you can associate it with a callback because your first extension can trigger it through evalScript and when the evalScript callback arrives, you'll know the extension has been opened (provided that it exists on the user's machine).
